distributed/scheduler.py

```diff
@@ -7325,7 +7544,7 @@ def _remove_from_processing(state: SchedulerState, ts: TaskState) -> str | None:
     assert ws
     ts.processing_on = None
 
-    if ws.address not in state.workers:  # may have been removed
+    if state.workers.get(ws.address) is not ws:  # may have been removed
```

Comment:
Note that I'm not merging main into this yet because this driveby fix here conflicts with https://github.com/dask/distributed/pull/6884#discussion_r952990650, so we'll need to decide what to do there first. cc @hendrikmakait